lib/codegen-meta/src/base/types.rs

```diff
@@ -0,0 +1,247 @@
+//! This module predefines all the Cranelift scalar types.
+
+// Numbering scheme for value types:
+//
+// 0: Void
+// 0x01-0x6f: Special types
+// 0x70-0x7f: Lane types
+// 0x80-0xff: Vector types
+//
+// Vector types are encoded with the lane type in the low 4 bits and log2(lanes)
+// in the high 4 bits, giving a range of 2-256 lanes.
+static LANE_BASE: u8 = 0x70;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Bool {
+    /// 1-bit bool.
+    B1 = 1,
```

Comment:
Lane types such as `Bool` or `Int` use enums with explicit discriminators, so that the `bits` value  can be set without a match statement. One example of where this happens is in the `impl From<base_types::Bool> for LaneType` block in `cdsl/types.rs` :)